tensorflow_io/core/plugins/hdfs/hadoop_filesystem.cc

```diff
@@ -0,0 +1,919 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <functional>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <string>
+#if !defined(_MSC_VER)
+#include <dlfcn.h>
+#endif
+
+#include "absl/strings/str_cat.h"
+#include "absl/synchronization/mutex.h"
+#include "hdfs/hdfs.h"
+#include "tensorflow/c/tf_status.h"
+#include "tensorflow_io/core/plugins/file_system_plugins.h"
+// TODO: Restore logging.h
+#define TF_Log(...)
+#define TF_VLog(...)
+// #include "tensorflow/c/logging.h"
+// TODO: Restore logging.h
+
+namespace tensorflow {
+namespace io {
+namespace hdfs {
+
+void* LoadSharedLibrary(const char* library_filename, TF_Status* status) {
+#if defined(_MSC_VER)
+  TF_SetStatus(status, TF_UNIMPLEMENTED, "LoadSharedLibrary not implemented");
+  return nullptr;
+#else
```

Comment:
Is there any reason not to use `TF_LoadSharedLibrary` as we did in experiemental plugin. And does it mean that we drop support for `hdfs` on Windows ?

Reply:
@vnvo2409 Inside tensorflow while the functions are defined, they somehow are not part of the pip package so we cannot include them directly. I tried to modify the tensorflow repo but had some trouble due to the big circular dependency. Give we only need those two functions and they are small, I think it makes sense to just include them here.

Also, I have added the Windows counterpart as well so now hdfs should be supported on Windows.